scrapy/exceptions.py

```diff
@@ -11,6 +11,11 @@ class NotConfigured(Exception):
     """Indicates a missing configuration situation"""
     pass
 
+class InvalidValue(TypeError):
```

Comment:
I renamed the exception to `InvalidOutput`, but kept it public because of the reasons above. Also, I think subclassing from [`TypeError`](https://docs.python.org/3/library/exceptions.html#TypeError) (_"Passing arguments of the wrong type (e.g. passing a list when an int is expected) should result in a TypeError"_) is more appropriate than [`ValueError`](https://docs.python.org/3/library/exceptions.html#ValueError) (_"Raised when a built-in operation or function receives an argument that has the right type but an inappropriate value"_)